common/file/archive/zip_test.go

```diff
@@ -0,0 +1,75 @@
+package archive
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+var (
+	tempDir string
+)
+
+func TestMain(m *testing.M) {
+	var err error
+	tempDir, err = ioutil.TempDir("", "gct-temp")
+	if err != nil {
+		fmt.Printf("failed to create temp file: %v", err)
+		os.Exit(1)
+	}
+	t := m.Run()
+	err = os.RemoveAll(tempDir)
+	if err != nil {
+		fmt.Printf("Failed to remove temp db file: %v", err)
+	}
+	os.Exit(t)
+}
+
+func TestZip(t *testing.T) {
+	zipFile := filepath.Join("..", "..", "..", "testdata", "testdata.zip")
+	files, err := Unzip(zipFile, tempDir)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(files) != 2 {
+		t.Fatalf("expected 2 files to be extracted received: %v ", len(files))
+	}
+
+	zipFile = filepath.Join("..", "..", "..", "testdata", "zip-slip.zip")
+	_, err = Unzip(zipFile, tempDir)
+	if err == nil {
+		t.Fatal("Zip() expected to error due to ZipSlip detection but extracted successfully")
+	}
+}
+func TestUnZip(t *testing.T) {
+	singleFile := filepath.Join("..", "..", "..", "testdata", "configtest.json")
+	outFile := filepath.Join(tempDir, "out.zip")
+	err := Zip(singleFile, outFile)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = Unzip(outFile, tempDir)
```

Comment:
`_` can be tested to ensure that it contains the files you put into it
edit: looks like you test it in `TestZip()`, but you should at least test the unzipping once in `TestUnzip()`